Storage/src/Bucket.php

```diff
@@ -1242,6 +1243,101 @@ public function lockRetentionPolicy(array $options = [])
         );
     }
 
+    /**
+     * Create a Signed URL listing objects in this bucket.
+     *
+     * Example:
+     * ```
+     * $url = $bucket->signedUrl(time() + 3600);
+     * ```
+     *
+     * ```
+     * // Use V4 Signing
+     * $url = $bucket->signedUrl(time() + 3600, [
+     *     'version' => 'v4'
+     * ]);
```

Comment:
Potential Bug: longer than 7 day expiration can be signed.
1. I set expiration time to variable `$expiration = time() + 604800 + 10`.
2. Wait 11 seconds.. due to some processing.
3. Then call `signedUrl`
```
$url = $bucket->signedUrl(expiration, [
     'version' => 'v4'
]);
```

A signed URL is created for `604800 + 10` seconds.

Reply:
Hey @frankyn, I don't think we are affected by this bug. Let me demonstrate.

Using your example, I've exploded the query string out so we can inspect the pieces.

```php
$expiration = time() + 604800 + 10;
sleep(11);
$url = $object->signedUrl($expiration, [
    'version' => 'v4'
]);

$url = parse_url($url);
parse_str($url['query'], $query);
print_r($query);
```

Result:
```
Array
(
    <snip>
    [X-Goog-Expires] => 604799
    <snip>
)
```

I could see it possibly being a trip for users who didn't realize that their expiration which was valid when defined is not valid when used, but I'm unsure about how that can be mitigated without a backwards-incompatible break to move from absolute to relative expirations. However, I've not been able to actually create a signed URL with an invalid expiration date.

Please forgive me if I'm misunderstanding what you're driving at! :)

Reply:
False alarm. This is working as intended. Please disregard my comment.